bot/cogs/events.py

```diff
@@ -103,6 +104,30 @@ def mod_log(self) -> ModLog:
         resp = await response.json()
         return resp["data"]
 
+    async def has_active_mute(self, user_id: str) -> bool:
+        """
+        Check whether a user has any active mute infractions
+        """
+
+        response = await self.bot.http_session.get(
+            URLs.site_infractions_user.format(
+                user_id=user_id
+            ),
+            params={"hidden": "True"},
+            headers=self.headers
+        )
+        infraction_list = await response.json()
+
+        # Check for active mute infractions
+        if len(infraction_list) == 0:
+            # Short circuit
+            return False
+
+        muted_check = any(
+            [infraction["active"] for infraction in infraction_list if infraction["type"].lower() == "mute"]
```

Comment:
https://github.com/python-discord/bot/blob/d5cb479be6925a10cf0c46e3088518f56a5a91b3/bot/cogs/events.py#L126-L128

Since any accepts a generator, we don't need to build the list first with a list comp:

```suggestion
            infraction["active"] for infraction in infraction_list if infraction["type"].lower() == "mute"
```

Reply:
Oh, duh, that's much smarter :)